### ISSUE TITLE
Issue with `atomic-spans`

### ISSUE THREAD

**Issue by denialhaag** [rvben/rumdl#748]
### rumdl version

0.2.37

### Installation method

Pre-built binary

### Bug description

MD013 may fail for spans exceeding the configured line length even though `atomic-spans` is enabled.

### Steps to reproduce

Run `rumdl` on the following file:

```markdown
_This is a very, very, very, very, very, very, very long line that exceeds 80 characters with some `code` inside._
```

`rumdl` fails with:

```markdown
example.md:1:81: [MD013] Line length 114 exceeds 80 characters
```

### Expected behavior

It shouldn't fail because `atomic-spans` is enabled.

### Actual behavior

It does fail.

### Configuration

```toml
[MD013]
line-length = 80
atomic-spans = trie
code-blocks = false
headings = false
reflow = true
reflow-mode = "normalize"
```

### Operating system

macOS

### Additional context

From what I've seen, MD013 only fails if two "span types" are mixed (e.g., "_" and "`").

**Comment by rvben** [comment]
Fixed in [v0.2.39](https://github.com/rvben/rumdl/releases/tag/v0.2.39).

Thanks for the precise report - your observation that it only fails when two span types are mixed was exactly right, and it generalizes: any nested `` ` ``, `*`, `_` or `~` inside the outer span triggered it.

`atomic-spans = true` never meant "never break an emphasis span". A span that on its own exceeds `line-length` is still broken, because holding it whole leaves a line that can never fit. That escape hatch was being cancelled for any content containing an inline marker, so a span with nested markup was unbreakable at any length - MD013 then reported a violation its own fixer refused to touch. Worse, normalize mode re-joined a correctly hand-wrapped paragraph back into one over-long line, so `rumdl fmt` produced output that `rumdl check` rejected. This was a regression in 0.2.37; 0.2.36 handled your input correctly.

Such a span now wraps at whitespace outside any nested construct. Nested code spans, links, images and emphasis are each held as one unbreakable unit, so a break can never collapse a code span's literal whitespace or a link destination's structural whitespace. That last part also fixed an older bug where `[text](<a b c>)` inside an emphasis span could be split mid-destination.

**Comment by denialhaag** [comment]
Thanks a lot for the quick fix, @rvben! 🙂 

One small observation: This example still fails, likely for a related reason:

```markdown
***This is a very, very, very, very, very, very, very, very, very, very long line that is bold and italic.***
```

**Comment by rvben** [comment]
Reopening: `***...***` is the same class of bug, not a variant.

Confirmed. It is not limited to `***`. Every doubly-nested form fails to wrap: `***x***`, `___x___`, `**_x_**`, `*__x__*`, `**~~x~~**`. Single-level `**x**`, `*x*` and `~~x~~` all wrap correctly. In each failing case MD013 reports a violation and `rumdl fmt` is a no-op, so the fixer again refuses to touch a line the linter flags.

The cause is the part of the 0.2.39 fix that holds a nested **emphasis** span atomic. That was right for code spans (whitespace is literal) and for link destinations (whitespace is structural), but it is too broad for emphasis: whitespace inside a nested emphasis span is ordinary prose whitespace. With `***text***`, the whole content of the outer span *is* one nested emphasis span, so there is no breakable whitespace anywhere inside it and the "a span longer than `line-length` is still broken" escape hatch has nothing left to act on.

[`docs/md013.md`](https://github.com/rvben/rumdl/blob/main/docs/md013.md) currently lists emphasis in the never-break set, so the doc needs the same correction as the code.

**Comment by denialhaag** [comment]
Version 0.2.41 fixes the issue! Thanks a lot for always acting so quickly! 🙂

**Comment by rvben** [comment]
Confirmed fixed in v0.2.41, and thanks for verifying on your end. Closing.
